frontend/src/uploads/UploadContainer.js

```diff
@@ -59,10 +60,10 @@ const UploadContainer = () => {
           filename,
           datasetSelected,
           replace,
-        }).then((postResponse) => {
-          setAlertContent(`Data has been successfully uploaded. ${postResponse.data}`);
-          setAlertSeverity('success');
+        }).then((response) => {
           setAlert(true);
+          setAlertSeverity(response.data.success ? 'success' : 'error')
+          setAlertContent(`${response.data.message}${response.data.errors ? '\nErrors: ' + response.data.errors.join('\n') : ''}`);
```

Comment:
I think if there's an error (non-2XX status code) when calling `import_data`, the `catch` block below should handle the response. There's a `showError` function in this container that you can modify to handle any new types of errors that come up!